spring-cloud-data-rest-resource/src/main/java/org/springframework/cloud/data/rest/resource/ModuleRegistrationResource.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.rest.resource;
+
+import org.springframework.hateoas.PagedResources;
+import org.springframework.hateoas.ResourceSupport;
+
+/**
+ * Rest resource for a module registration.
+ *
+ * @author Glenn Renfro
+ * @author Mark Fisher
+ * @author Patrick Peralta
+ */
+public class ModuleRegistrationResource extends ResourceSupport {
+
+	/**
+	 * Module name.
+	 */
+	private String name;
+
+	/**
+	 * Module type.
+	 */
+	private String type;
+
+	/**
+	 * Maven coordinates for module artifact. String is in standard format
+	 * such as {@code groupId:artifactId:version}.
+	 */
+	private String coordinates;
+
+	/**
+	 * True if this is a composed module.
+	 */
+	private boolean composed;
+
+	/**
+	 * Default constructor for serialization frameworks.
+	 */
+	protected ModuleRegistrationResource() {
+	}
+
+	/**
+	 * Construct a {@code ModuleRegistrationResource}.
+	 *
+	 * @param name module name
+	 * @param type module type
+	 * @param coordinates coordinates for module artifact
+	 * @param composed true if this is a composed module
```

Comment:
Are we going to have `composed` module at the ModuleRegistration level? If I understand it correct, it will be the responsibility of the module author to compose modules together (using aggregate builder pattern) and make it pointing to a single maven co-ordinate. For the modules with different maven co-ordinates, we can do the module composition using `direct-binding` while still have it connected to other modules via binder.